pandas/tests/series/methods/test_duplicated.py

```diff
@@ -50,3 +51,15 @@ def test_duplicated_categorical_bool_na(nulls_fixture):
     result = ser.duplicated()
     expected = Series([False, False, True, True, False])
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "keep, vals",
+    [("last", [True, False]), ("first", [False, True]), (False, [True, True])],
+)
+def test_duplicated_mask(keep, vals):
+    # GH#48150
+    ser = Series([1, 2, NA, NA], dtype="Int64")
```

Comment:
Nit: Could you add one more `NA` at the end? IIRC I think technically the last

```
                    else:
                        out[i] = 1
```
Doesn't have coverage with this test

Reply:
Good point, thx. Added

Reply:
But we should probably also cover the case when `out[i]` does not get set, i.e. there is only a single NA?

Reply:
thx, this found a bug